### DISCUSSION TITLE
eslint-plugin-import not working with Bit

### DISCUSSION THREAD

**Discussion by dreamawakening** [teambit/bit#4636] [Help]
I use `eslint-plugin-import` to verify packages I'm importing exist/are correct. It checks `package.json` to see if a dependency exists. See [this rule](https://github.com/konraddroeske/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md) for details.

However, because I'm installing dependencies with `bit install`, it means dependencies aren't populated in `package.json`. Therefore I can't use `eslint-plugin-import`. I don't want to disable it just so I can use Bit.

Is there a way to get this to work with Bit?

**Comment by itaymendel** [comment]
As this rule requires a `package.json` at the base of your repository, you will need to have one.

You can manage dependencies in a `package.json` instead of using `workspace.jsonc`. Bit will read dependencies from there if needed. However, there are a couple of things to know:

1. Add dependencies with `npm i` / `yarn add` directly, not with `bit install`.
2. Add `bit link` to you `post install` for Bit to generate your component-dists in `node_modules` (package managers sometimes remove directories they don't manage from `node_modules`).
3. Ensure no dependencies are listed in your `workspace.json`.
4. You can still use `bit install` to install deps, as Bit propagates to your `package.json` for installing all your components.

If you do take this route, you loose Bit's automated "workspace" management for dependencies, where if different components require different versions of the same dependency, Bit automates [Yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) or [pnpm workspaces](https://pnpm.io/workspaces) (and soon [npm workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces)) behind the scenes for this to work seamlessly.

**Reply by dreamawakening** [reply]
@itaymendel I do have a `package.json` file. However, I don't manage my dependencies in here with Yarn due to the limitations you mentioned. I read that in the docs earlier.

Instead, I use `bit install`. However, this doesn't populate `package.json`, meaning it doesn't work with `eslint-plugin-import` (for the reasons I described).

Additionally, there's another problem. When I import my own components inside a component, such as:

```ts
import "@cannabitrx/tailwind.config";
```

`eslint-plugin-import` won't pick up on this either. Because `@cannabitrx/tailwind.config` isn't in my `node_modules`, and it's also not listed in the `package.json`. So it doesn't work for this case, either.

**Comment by GiladShoham** [comment]
see also my comments [here](https://github.com/import-js/eslint-plugin-import/issues/2096#issuecomment-909058777)
